parser.go

```diff
@@ -743,34 +727,27 @@ loop:
 			break loop
 		}
 		if err != nil {
-			return nil, err
+			return result, err
 		}
-		if result == nil {
-			result = ns
-		} else {
-			result = intersectionSelector(result, ns)
-		}
-	}
 
-	if result == nil {
-		result = func(n *html.Node) bool {
-			return n.Type == html.ElementNode
-		}
+		result.selectors = append(result.selectors, ns)
 	}
-
 	return result, nil
 }
 
 // parseSelector parses a selector that may include combinators.
-func (p *parser) parseSelector() (result Selector, err error) {
+func (p *parser) parseSelector() (result Matcher, err error) {
```

Comment:
Here is a case where you need to detect that there is only a single selector in the group, and return that selector rather than the group.